solver/exec.go

```diff
@@ -0,0 +1,109 @@
+package solver
+
+import (
+	"context"
```

Comment:
unrelated to this PR itself, but I'd avoid importing both golang.org/x/net/context and context